osu.Game/Scoring/ScoreImporter.cs

```diff
@@ -182,6 +182,12 @@ protected override void PostImport(ScoreInfo model, Realm realm, ImportParameter
             base.PostImport(model, realm, parameters);
 
             populateUserDetails(model);
+
+            Debug.Assert(model.BeatmapInfo != null);
+
+            // This needs to be run after user detail population to ensure we have a valid user id.
+            if (api.IsLoggedIn && api.LocalUser.Value.OnlineID == model.UserID && (model.BeatmapInfo.LastPlayed == null || model.Date > model.BeatmapInfo.LastPlayed))
```

Comment:
I guess this is accepting that due to how the replay format works, username changes can potentially cause the score to be misattributed to the local user and thus also change the last played date?

I guess it's not a huge deal but I do wanna ask to make sure we want to keep using this piece of potentially-unreliable data for more stuff.

Reply:
There's an assumption the flow here will be replaced / get more reliable going forward.